File: src/dataset/cpp/difference_of_squares_Exercism_thefullarcticfox/gemini/GeminiFlash_difference_of_squares_v2.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

long square_of_sum(long n) {
  long sum = n * (n + 1) / 2;
  return sum * sum;
}

long sum_of_squares(long n) {
  return n * (n + 1) * (2 * n + 1) / 6;
}

long difference(long n) {
  long sum_of_sq = n * (n + 1) * (2 * n + 1) / 6;
  long sum = n * (n + 1) / 2;
  long square_of_sum_val = sum * sum;
  return square_of_sum_val - sum_of_sq;
}

}  // namespace difference_of_squares

File: src/dataset/typescript/difference-of-squares_Exercism-typescript-alemarr/gemini/GeminiFlash_difference-of-squares_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Squares = void 0;
class Squares {
    constructor(number) {
        this.number = number;
        this.sum = (number * (number + 1)) / 2;
        this.sumOfSquaresValue = (number * (number + 1) * (2 * number + 1)) / 6;
    }
    get sumOfSquares() {
        return this.sumOfSquaresValue;
    }
    get squareOfSum() {
        return this.sum * this.sum;
    }
    get difference() {
        return this.squareOfSum - this.sumOfSquares;
    }
}
exports.Squares = Squares;
